eva-pipeline/src/test/java/embl/ebi/variation/eva/pipeline/steps/VariantAnnotLoadBatchTest.java

```diff
@@ -24,17 +47,127 @@
  */
 public class VariantAnnotLoadBatchTest {
 
+
+    private VariantAnnotLoadBatch variantAnnotLoadBatch;
+    private ExecutionContext executionContext;
+
+    @Before
+    public void setUp() throws Exception {
+        executionContext = MetaDataInstanceFactory.createStepExecution().getExecutionContext();
+
+        variantAnnotLoadBatch = new VariantAnnotLoadBatch();
+    }
+
+    @Test
+    public void variantAnnotationReaderShouldReadAllFieldsInVepOutput() throws Exception {
+        VariantAnnotationLineMapper lineMapper = new VariantAnnotationLineMapper();
+        for (String annotLine : vepOutput.split("\n")) {
+            VariantAnnotation variantAnnotation = lineMapper.mapLine(annotLine, 0);
+            assertTrue(variantAnnotation.getConsequenceTypes() != null);
+        }
+    }
+
     @Test
-    public void variantAnnotationReaderShouldReadAllFieldsInVepOutput(){
+    public void variantAnnotationReaderShouldReadAllLinesInVepOutput() throws Exception {
+        String vepOutput = VariantAnnotLoadBatchTest.class.getResource("/annot.tsv.gz").getFile();
+
+        FlatFileItemReader<VariantAnnotation> reader = variantAnnotLoadBatch.initReader(new GzipLazyResource(vepOutput));
+
+        reader.setSaveState(false);
+        reader.open(executionContext);
+
+        VariantAnnotation variantAnnotation;
+        int consequenceTypeCount = 0;
+//        int otherCount = 0;
+        int count = 0;
+        while ((variantAnnotation = reader.read()) != null) {
+            count++;
+            if (variantAnnotation.getConsequenceTypes() != null && !variantAnnotation.getConsequenceTypes().isEmpty()) {
+                consequenceTypeCount++;
+            }
+        }
+        // all should have at least consequence type annotations
```

Comment:
Why are we testing this here? Should not be tested during annotation creation step? Here we just reading the annotation that we suppose to be correct. The test should be for checking that all lines are read (and not to check that the lines are correct==confusing)
if some lines are to be skipped then we need to create a processor: from doc.. One typical use for an item processor is to filter out records before they are passed to the ItemWriter. Filtering is an action distinct from skipping; skipping indicates that a record is invalid whereas filtering simply indicates that a record should not be written.